### PR TITLE
Update dependencies: tidy, helm, controller-runtime

### PR DESCRIPTION
**Pull request by pacevedom**
Changes:
- Tidy up dependencies.
- Bump helm to 3.6.3.
- Bump controller-runtime to 0.10.2.
- Bump k8s.io deps to 0.22.2.

### CODE REVIEW

**Comment by dagrayvid** [comment]
Thanks @pacevedom. As discussed on Slack I think we should bump the k8s dependencies to 0.22.0 and also bump the openshift dependencies. 

In the meantime, I will run the e2e test on these changes.

**Comment by dagrayvid** [comment]
The go-lint and make verify errors will be resolved by running `make patch` and updating the PR

**Comment by dagrayvid** [comment]
/hold

The Helm getters from the patch don't seem to work with the new helm version.

**Comment by dagrayvid** [comment]
/cc @zvonkok 

PTAL

**Comment by bthurber** [comment]
@pacevedom do we know next steps on this?

File: go.mod [hunk 1]
```diff
@@ -18,12 +18,12 @@ require (
...
-	helm.sh/helm/v3 v3.6.0
-	k8s.io/api v0.21.1
-	k8s.io/apimachinery v0.21.1
-	k8s.io/cli-runtime v0.21.0
-	k8s.io/client-go v0.21.1
+	helm.sh/helm/v3 v3.6.3
+	k8s.io/api v0.22.2
+	k8s.io/apimachinery v0.22.2
+	k8s.io/cli-runtime v0.22.2
+	k8s.io/client-go v0.22.2
```
Comment (qbarrand) [inline go.mod:25]: Why 0.22? I think operator-sdk's current version officially works with 0.21

**Comment by pacevedom** [comment]
Needs PR #122 before being testable.
One of the vendored libraries requires `io/fs` library which is only available in go 1.16. Workflow uses 1.15 and it is running the configuration from master branch.
Background: https://github.community/t/github-action-workflow-is-executed-for-a-pr-from-a-forked-repo-if-the-pr-changes-the-workflow-file/16059

File: go.mod (hunk 1 above)
Comment (pacevedom) [reply go.mod:25]: Version 1.14 is using 0.22.2 except for the cli-runtime, which is 0.21.0. Also OCP 4.9 is based on k8s 1.22.
WIll change this to align cli-runtime though.

**Pull request merged**